Services/Simulation/JavascriptInterpreter.cs

```diff
@@ -53,19 +53,28 @@ public class JavascriptInterpreter : IJavascriptInterpreter
             // logging into the service logs
             engine.SetValue("log", new Action<object>(this.JsLog));
 
-            //register callback for state updates
-            engine.SetValue("updateState", new Action<JsValue>(this.UpdateState));
+            // Register callback for state updates
+            engine.SetValue("updateState", new Action<JsValue>(this.JsMergeState));
+            engine.SetValue("reportMethodProgress", new Action<string>(this.JsReportMethodProgress));
 
-            //register sleep function for javascript use
-            engine.SetValue("sleep", new Action<int>(this.Sleep));
+            // Register sleep function for javascript use
+            engine.SetValue("sleep", new Action<int>(this.JsSleep));
+
+            // Register functions to enable/disable internal state simulation
+            engine.SetValue("enableSensorSimulation", new Action(this.JsEnableSensorSimulation));
```

Comment:
I pictured us turning sensor simulation off at an individual sensor level, not for all sensors.